kedro-datasets/kedro_datasets/spark/spark_streaming_dataset.py

```diff
@@ -0,0 +1,154 @@
+"""SparkStreamingDataSet to load and save a PySpark Streaming DataFrame."""
+from copy import deepcopy
+from pathlib import PurePosixPath
+from typing import Any, Dict
+
+from kedro.io.core import AbstractDataSet
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.utils import AnalysisException
+
+from kedro_datasets.spark.spark_dataset import (
+    SparkDataSet,
+    _split_filepath,
+    _strip_dbfs_prefix,
+)
+
+
+class SparkStreamingDataSet(AbstractDataSet):
+    """``SparkStreamingDataSet`` loads data into Spark Streaming Dataframe objects.
+    Example usage for the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+    .. code-block:: yaml
+        raw.new_inventory:
+        type: streaming.extras.datasets.spark_streaming_dataset.SparkStreamingDataSet
+        filepath: data/01_raw/stream/inventory/
+        file_format: json
+        save_args:
+            output_mode: append
+            checkpoint: data/04_checkpoint/raw_new_inventory
+            header: True
+        load_args:
+            schema:
+            filepath: data/01_raw/schema/inventory_schema.json
+    """
```

Comment:
Would it be possible to add a Python API example? Don't worry if it's too much work.

More importantly the `schema` doesn't look right, I am looking for something I can quickly paste on a terminal and run.

Reply:
We have tested python API example, we have updated the schema